pandas/_libs/missing.pyx

```diff
@@ -160,7 +161,7 @@ def isnaobj_old(arr: ndarray) -> ndarray:
     result = np.zeros(n, dtype=np.uint8)
     for i in range(n):
         val = arr[i]
-        result[i] = val is NaT or _check_none_nan_inf_neginf(val)
+        result[i] = checknull(val) or val == INF or val == NEGINF
```

Comment:
Is this related to the OP in #33594?. I _think_ the NA checking for string dtype https://github.com/pandas-dev/pandas/issues/33594#issuecomment-615879369 maybe should be a separate issue/PR. We _may_ want to backport the fix relating to the change in return type of Categorical.ravel from numpy array to Categorical.

Reply:
It's unrelated, I found this other bug while debugging the original issue and decided to fix it here. I'll go ahead and separate it out for easier bookkeeping.

Reply:
Moved to https://github.com/pandas-dev/pandas/pull/33656

Reply:
Actually, they are kind of related in that a fix needs to be made missing.py as well. I'll wait to see how that ends up here before making the associated change in the other PR.

Reply:
one is a bug in `_isna_ndarraylike_old` and the other is a bug in `isnaobj_old`. They are related but I think should be treated indepedently.

for the NA issue (bug in `isnaobj_old` once bug in `_isna_ndarraylike_old` is fixed) we may want to add more tests for other extension arrays or add a test/parameterise an existing test in the base extension array tests.